source/Body.hpp

```diff
@@ -49,24 +54,29 @@ using namespace std;
 namespace moordyn {
 
 class Waves;
+typedef std::shared_ptr<Waves> WavesRef;
```

Comment:
This is a bit ugly... I really like predeclarations, but here it is way better to move the Waves.hpp include here, so we do not need to redefine WavesRef.